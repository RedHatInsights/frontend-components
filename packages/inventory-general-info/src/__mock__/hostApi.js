import { hosts } from '../api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios);
