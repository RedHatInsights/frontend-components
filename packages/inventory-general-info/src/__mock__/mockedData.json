{
    "total": 1,
    "count": 1,
    "page": 1,
    "per_page": 50,
    "results": [
        {
            "id": "6de30d78-f052-41d4-b123-125017443a98",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "7.4",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu"
            }
        }
    ]
}
