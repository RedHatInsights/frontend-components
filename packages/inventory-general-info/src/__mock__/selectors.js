/* eslint-disable camelcase */
export const testProperties = {
    number_of_cpus: 1,
    number_of_sockets: 1,
    cores_per_socket: 1,
    ramSize: '5 MB',
    disk_devices: []
};

export const osTest = {
    arch: 'test-arch',
    os_release: 'test-release',
    os_kernel_version: 'test-kernel',
    last_boot_time: 'test-boot',
    kernel_modules: []
};

export const biosTest = {
    bios_vendor: 'test-vendor',
    bios_version: 'test-version',
    bios_release_date: 1565702893431,
    cpu_flags: []
};

export const infraTest = {
    infrastructure_type: 'test-type',
    infrastructure_vendor: 'test-vendor',
    network: {
        ipv4: [ '1' ],
        ipv6: [ '6' ],
        interfaces: [ 'test' ]
    }
};

export const configTest = {
    installed_packages: [ 'packages' ],
    enabled_services: [ 'services' ],
    running_processes: [ 'processes' ],
    repositories: {
        enabled: [{
            base_url: 'test-url',
            name: 'test-name',
            enabled: true,
            gpgcheck: true
        }],
        disabled: []
    }
};

export const collectInfoTest = {
    insights_client_version: 'test-client',
    insights_egg_version: 'test-egg'
};
