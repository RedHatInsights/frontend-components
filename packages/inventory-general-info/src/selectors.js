/* eslint-disable camelcase */

function safeParser(toParse, key) {
    try {
        return JSON.parse(toParse);
    } catch {
        return { [key]: toParse };
    }
}

export const propertiesSelector = ({
    number_of_cpus,
    number_of_sockets,
    cores_per_socket,
    ramSize,
    disk_devices
} = {}) => ({
    cpuNumber: number_of_cpus,
    sockets: number_of_sockets,
    coresPerSocket: cores_per_socket,
    ramSize,
    storage: disk_devices && disk_devices.map(({ device, label, mount_point, options, type }) => ({
        ...device && safeParser(device, 'device'),
        label,
        ...mount_point && safeParser(mount_point, 'mountpoint'),
        ...options && safeParser(options, 'options'),
        ...type && safeParser(type, 'mounttype')
    })
    )
});

export const operatingSystem = ({
    arch,
    os_release,
    os_kernel_version,
    last_boot_time,
    kernel_modules
} = {}) => ({
    release: os_release,
    kernelRelease: os_kernel_version,
    architecture: arch,
    bootTime: last_boot_time,
    kernelModules: kernel_modules
});

export const biosSelector = ({
    bios_vendor,
    bios_version,
    bios_release_date,
    cpu_flags
} = {}) => ({
    vendor: bios_vendor,
    version: bios_version,
    releaseDate: bios_release_date && new Date(bios_release_date).toLocaleDateString(),
    csm: cpu_flags
});

export const infrastructureSelector = ({
    infrastructure_type,
    infrastructure_vendor,
    network = {}
} = {}) => ({
    type: infrastructure_type,
    vendor: infrastructure_vendor,
    ipv4: network.ipv4,
    ipv6: network.ipv6,
    nics: network.interfaces
});

export const configurationSelector = ({
    installed_packages,
    enabled_services,
    running_processes,
    repositories
} = {}) => ({
    packages: installed_packages,
    services: enabled_services,
    processes: running_processes,
    repositories
});

export const collectionInformationSelector = ({
    insights_client_version,
    insights_egg_version
} = {}) => ({
    client: insights_client_version,
    egg: insights_egg_version
});
