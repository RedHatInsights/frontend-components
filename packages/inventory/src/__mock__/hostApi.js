import { hosts, tags } from '../api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios);
export const mockTags = new MockAdapter(tags.axios);
