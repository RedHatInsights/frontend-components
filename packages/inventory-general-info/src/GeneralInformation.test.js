/* eslint-disable camelcase */
import React from 'react';
import { render, mount, shallow } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import GeneralInformation from './GeneralInformation';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { osTest, biosTest, collectInfoTest, configTest, infraTest, testProperties } from './__mock__/selectors';
import promiseMiddleware from 'redux-promise-middleware';
import { mock } from './__mock__/hostApi';
import mockedData from './__mock__/mockedData.json';

describe('GeneralInformation', () => {
    let initialState;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore([ promiseMiddleware() ]);
        initialState = {
            entityDetails: {
                entity: {
                    id: 'test-id'
                }
            },
            systemProfileStore: {
                systemProfile: {
                    loaded: true,
                    ...infraTest,
                    ...osTest,
                    ...biosTest,
                    ...collectInfoTest,
                    ...configTest,
                    ...testProperties,
                    network: {
                        ipv4: [ '1', '2' ],
                        ipv6: [ '6', '3' ],
                        interfaces: [{
                            mac_address: '0:0:0',
                            mtu: 150,
                            name: 'some name',
                            state: 'UP',
                            type: 'some type'
                        }, {
                            mac_address: '1:0:0',
                            mtu: 1150,
                            name: 'asome name',
                            state: 'UP',
                            type: 'asome type'
                        }]
                    }
                }
            }
        };
    });

    it('should render correctly - no data', () => {
        const store = mockStore({ systemProfileStore: {}, entityDetails: {} });
        const wrapper = render(<Provider store={ store }>
            <GeneralInformation />
        </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = render(<Provider store={ store }>
            <GeneralInformation />
        </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('API', () => {
        mock.onGet('/api/inventory/v1/hosts/test-id/system_profile').reply(200, mockedData);
        it('should get data from server', () => {
            const store = mockStore({
                systemProfileStore: {},
                entityDetails: {
                    entity: {
                        id: 'test-id'
                    }
                } });
            mount(<Provider store={ store }>
                <GeneralInformation />
            </Provider>);
            expect(store.getActions()[0].type).toBe('LOAD_SYSTEM_PROFILE_PENDING');
        });

        it('should open modal', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <GeneralInformation />
            </Provider>);
            wrapper.find('a[href$="interfaces"]').first().simulate('click');
            wrapper.update();
            expect(wrapper.find('GeneralInformation').instance().state.isModalOpen).toBe(true);
            expect(wrapper.find('GeneralInformation').instance().state.modalTitle).toBe('Interfaces/NICs');
        });

        it('should update on sort', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <GeneralInformation />
            </Provider>);
            wrapper.find('a[href$="interfaces"]').first().simulate('click');
            wrapper.update();
            const [ firstRow, secondRow ] = wrapper.find('GeneralInformation').instance().state.rows;
            wrapper.find('table th button').first().simulate('click');
            wrapper.update();
            expect(wrapper.find('GeneralInformation').instance().state.rows[0]).toEqual(secondRow);
            expect(wrapper.find('GeneralInformation').instance().state.rows[1]).toEqual(firstRow);
        });

        it('should open modal', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <GeneralInformation />
            </Provider>);
            wrapper.find('a[href$="interfaces"]').first().simulate('click');
            wrapper.update();
            wrapper.find('button.pf-m-plain').first().simulate('click');
            wrapper.update();
            expect(wrapper.find('GeneralInformation').instance().state.isModalOpen).toBe(false);
        });

        it('should calculate first index when expandable', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <GeneralInformation />
            </Provider>);
            wrapper.find('GeneralInformation').instance().handleModalToggle('title', {
                cells: [{ title: 'one' }, { title: 'two' }],
                rows: [
                    { cells: [ 'a', 'aa' ] },
                    { cells: [ 'b', 'bb' ] }],
                expandable: true
            });
        });
    });
});
