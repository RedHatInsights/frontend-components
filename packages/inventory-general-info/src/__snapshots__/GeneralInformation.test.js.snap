// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralInformation should render correctly - no data 1`] = `
<div
  class="pf-l-grid pf-m-all-12-col-on-sm pf-m-all-6-col-on-md pf-m-gutter"
>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            System properties
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Host name
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__md"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Display name
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__md"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Ansible hostname
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__md"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Number of CPUs
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Sockets
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Cores per socket
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              RAM
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Storage
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Operating System
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Release
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Kernel release
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Architecture
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Last boot time
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Kernel modules
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            BIOS
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Vendor
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Version
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Release date
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              CSM
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Type
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Vendor
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Installed packages
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Services
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Running processes
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Repositories
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Collection information
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Insights client
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Egg
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Last check-in
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Registered
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <div
                class="ins-c-skeleton ins-c-skeleton__sm"
              >
                 
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GeneralInformation should render correctly 1`] = `
<div
  class="pf-l-grid pf-m-all-12-col-on-sm pf-m-all-6-col-on-md pf-m-gutter"
>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            System properties
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Host name
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              Not Available
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Display name
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                class="ins-c-inventory__detail--action"
                href="http://localhost:5000//display_name"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align:-0.125em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                    transform=""
                  />
                </svg>
              </a>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Ansible hostname
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-id
              <a
                class="ins-c-inventory__detail--action"
                href="http://localhost:5000//ansible_name"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align:-0.125em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                    transform=""
                  />
                </svg>
              </a>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Number of CPUs
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              1
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Sockets
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              1
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Cores per socket
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              1
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              RAM
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              5 MB
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Storage
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//storage"
              >
                0 disks
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Operating System
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Release
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-release
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Kernel release
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-kernel
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Architecture
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-arch
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Last boot time
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              Invalid Date
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Kernel modules
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//kernel_modules"
              >
                0 modules
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            BIOS
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Vendor
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-vendor
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Version
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-version
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Release date
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              8/13/2019
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              CSM
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              None
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Infrastructure
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Type
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-type
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Vendor
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-vendor
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              IPv4 addresses
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//ipv4"
              >
                2 addresses
              </a>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              IPv6 addresses
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//ipv6"
              >
                2 addresses
              </a>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Interfaces/NICs
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//interfaces"
              >
                2 NICs
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Configuration
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Installed packages
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//installed_packages"
              >
                1 packages
              </a>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Services
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//services"
              >
                1 services
              </a>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Running processes
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//running_processes"
              >
                1 processes
              </a>
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Repositories
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              <a
                href="http://localhost:5000//repositories"
              >
                1 enabled
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-l-grid__item"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Collection information
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
            data-pf-content="true"
          >
            <dt
              class=""
              data-pf-content="true"
            >
              Insights client
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-client
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Egg
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              test-egg
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Last check-in
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              Invalid Date
            </dd>
            <dt
              class=""
              data-pf-content="true"
            >
              Registered
            </dt>
            <dd
              class=""
              data-pf-content="true"
            >
              Invalid Date
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</div>
`;
